inc/class-sitemaps-index.php

```diff
@@ -45,37 +53,82 @@ public function redirect_canonical( $redirect ) {
 	}
 
 	/**
-	 * Produce XML to output.
+	 * Get all of the available sitemaps from the registry
+	 * and add to an array.
+	 *
+	 * @todo get_registered_sitemaps() and get_sitemap_urls() are looping through teh array and
+	 * nested array to get at the value for ['route']. There is probably a better way to do
+	 * this than two methods that are almost identical.
+	 *
+	 * @return array $sitemaps_list
 	 */
-	public function render_sitemap() {
-		$sitemap_index = get_query_var( 'sitemap' );
+	public function get_registered_sitemaps() {
+		$sitemaps_list = array();
+		$sitemaps_all = $this->registry->get_sitemaps();
 
-		if ( 'sitemap_index' === $sitemap_index ) {
-			header( 'Content-type: application/xml; charset=UTF-8' );
+		foreach ( $sitemaps_all as $sitemaps ) {
+			array_push( $sitemaps_list, $sitemaps );
+		}
 
-			echo '<?xml version="1.0" encoding="UTF-8" ?>';
-			echo '<sitemapindex xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">';
+		return $sitemaps_list;
+	}
 
-			echo '</sitemapindex>';
-			exit;
+	/**
+	 * Get all of the URLs for the sitemaps and add to an array.
+	 *
+	 * @return array $sitemaps_urls
+	 */
+	public function get_sitemap_urls() {
+		$sitemap_urls = array();
+		$sitemaps_list = $this->get_registered_sitemaps();
+
+		foreach ( $sitemaps_list as $sitemap ) {
+			array_push( $sitemap_urls, $sitemap );
 		}
+
+		return $sitemap_urls;
 	}
 
 	/**
-	 * Builds the URL for the sitemap index.
+	 * Add the correct xml to any given url.
+	 *
+	 * @todo This will also need to be updated with the last modified information as well.
 	 *
-	 * @return string the sitemap index url.
+	 * @return string $markup
 	 */
-	public function sitemap_index_url() {
-		global $wp_rewrite;
+	public function get_index_url_markup( $url ) {
+		$markup = '<sitemap>' . "\n";
+		$markup .= '<loc>' . $url . '</loc>' . "\n";
```

Comment:
```suggestion
		$markup .= '<loc>' . esc_url($url) . '</loc>' . "\n";
```
The $url is potentially unsafe and could contain >< or other characters.

Reply:
yeah, sorry this is another small thing I updated while you were reviewing